src/base/worker_pool.h

```diff
@@ -10,18 +10,27 @@ namespace base {
 
 class WorkerPool {
  public:
-  using NetWorker = Fn<void(const Atomic<bool>&, Data&)>;
-  using SimpleWorker = Fn<void(const Atomic<bool>&)>;
+  using NetWorker = Fn<void(WorkerPool*, Data&)>;
+  using SimpleWorker = Fn<void(WorkerPool*)>;
 
   explicit WorkerPool(bool force_shut_down = false);
   ~WorkerPool();
 
   void AddWorker(Literal name, const NetWorker& worker, ui32 count = 1);
   void AddWorker(Literal name, const SimpleWorker& worker, ui32 count = 1);
 
+  bool WaitUntilShutdown(
+      const std::chrono::duration<double, std::milli>& duration);
```

Comment:
Why `milli`? Why not `seconds`? - You can't specify anywhere millis, i.e. in config.